dao-macros/src/contract/contract_test.rs

```diff
@@ -75,19 +58,55 @@ fn generate_test_implementation(input: &CasperContractItem) -> TokenStream {
                 assert_eq!(self.event::<T>(index), event);
             }
         }
-    }
+    })
 }
 
-fn generate_test_interface(input: &CasperContractItem) -> TokenStream {
+fn build_constructor(item: &CasperContractItem) -> Result<TokenStream, syn::Error> {
+    let init = match utils::find_method(item, "init") {
+        Some(method) => method,
+        None => {
+            return Err(syn::Error::new(
+                Span::call_site(),
+                "Contract has to define init() method",
+            ))
+        }
+    };
+    let casper_args = utils::generate_method_args(init);
+    let mut args: Punctuated<FnArg, Comma> = Punctuated::new();
+    init.sig
+        .clone()
```

Comment:
You should be able to clone `inputs`. So instrad it would be `init.sig.inputs.clone()`.